### PR TITLE
[CNFT2-1600] [CNFT2-1604] - Page_Info and Page History

### PR DESCRIPTION
**Pull request by lramakr**
## Description

[CNFT2-1600] [CNFT2-1604] - Developed Page_Info and Page History

## Tickets

* [Jira Ticket] https://cdc-nbs.atlassian.net/browse/CNFT2-1600, https://cdc-nbs.atlassian.net/browse/CNFT2-1604

## Checklist before requesting a review
- [x] PR focuses on a single story
- [x] Code has been fully tested to meet acceptance criteria
- [x] PR is reasonably small and reviewable (Generally less than 10 files and 500 changed lines)
- [x] All new functions/classes/components reasonably small
- [x] Functions/classes/components focused on one responsibility
- [x] Code easy to understand and modify (clarity over concise/clever)
- [x] PRs containing TypeScript follow the [Do's and Don'ts](https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html)
- [x] PR does not contain hardcoded values (Uses constants)
- [x] All code is covered by unit or feature tests

[CNFT2-1600]: https://cdc-nbs.atlassian.net/browse/CNFT2-1600?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CNFT2-1604]: https://cdc-nbs.atlassian.net/browse/CNFT2-1604?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx [hunk 1]
```diff
@@ -1,27 +1,165 @@
...
 const PageInformation = () => {
+    const [activeTab, setActiveTab] = useState('Details');
+    const [totalResults] = useState(4);
+    const [currentPage, setCurrentPage] = useState(1);
+    const [pageHistory, setPageHistory] = useState([
+        { publishVersionNbr: '', lastUpdatedDate: '', lastUpdatedBy: '', notes: '' }
+    ]);
+    const [pageInfo, setPageInfo] = useState<InfoType>({});
+    const { pageId } = useParams();
+    const { page } = usePageManagement();
```
Comment (rkrusselenq) [inline apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx:25]: can you can the pageId from here?

File: apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx [hunk 2]
```diff
@@ -1,27 +1,165 @@
...
+    const pageSize = 10;
+    const navigate = useNavigate();
+
+    const fetchPageHistory = async () => {
+        const rep: any = await PageInformationService?.getPageHistory?.({
```
Comment (rkrusselenq) [inline apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx:30]: issue: use of any

File: apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx [hunk 3]
```diff
@@ -1,27 +1,165 @@
...
+        });
+        setPageHistory(rep);
+    };
+    const fetchPageInfo = () => {
+        PageInformationService.find({
+            authorization: authorization(),
+            page: Number(pageId)
+        }).then((data: InfoType) => {
+            setPageInfo(data);
+        });
```
Comment (rkrusselenq) [inline apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx:42]: need to .catch()

**Review by rkrusselenq — APPROVED** [review]

**Review by mpeels — CHANGES_REQUESTED** [review]

File: apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx [hunk 4]
```diff
@@ -1,27 +1,165 @@
...
+} from 'apps/page-builder/generated';
+import { download } from '../../../../../../utils/download';
+import React, { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+import { usePageManagement } from '../../usePageManagement';
+import { useNavigate } from 'react-router-dom';
 
 const PageInformation = () => {
+    const [activeTab, setActiveTab] = useState('Details');
+    const [totalResults] = useState(4);
```
[the remark this answers is not in the document]
Comment (lramakr) [reply apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx]: Looks like the page-history api missing the pagination details like total no of pages etc.. currently. I tried replicating it by hard coded values. Once the API is fixed, then I will revisit for pagination implementation

File: apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx [hunk 5]
```diff
@@ -1,27 +1,167 @@
...
+                History
+            </li>
+        </ul>
+    );
+    const renderBlock = (header: string, desc: string) => (
+        <div className={styles.lineBlock}>
+            <div className={styles.detailHeader}>{header}</div>
+            <div className={styles.smallBodyText}>{desc ?? '-'}</div>
+        </div>
+    );
```
Comment (mpeels) [inline apps/modernization-ui/src/apps/page-builder/page/management/preview/information/PageInformation.tsx:86]: Suggestion: `desc` is a required input but there is a check to verify it is not `undefined` (`desc ?? '-'`). I would suggest defining  `desc` as:

```typescript
desc?: string
```

Then you can eliminate the `non-null assertion operator` from:  `pageInfo?.eventType?.name!` and other `renderBlock` calls. The use of the `non-null assertion operator` is typically considered a bad practice as this simply overrides the compilers warning without providing a solution.

**Review by mpeels — APPROVED** [review]

**Review by rkrusselenq — APPROVED** [review]

**Pull request merged**